pkg/apis/devices/v1alpha2/zz_generated.register.go

```diff
@@ -0,0 +1,69 @@
+/*
+Copyright The KubeEdge Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Code generated by register-gen. DO NOT EDIT.
+
+package v1alpha2
+
+import (
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+// GroupName specifies the group name used to register the objects.
+const GroupName = "devices.kubeedge.io"
```

Comment:
what's GroupName generated from？how `hack/update-codegen.sh` construct this groupname with the suffix `.kubeedge.io`

Reply:
We set it in the comment of [doc.go](https://github.com/kubeedge/kubeedge/pull/4017/files#diff-21f741b214c9d2a115d1ba0a8aba09960ad3c6d6fd4f7ee261a87ffecee02d6bR22). It's originally at [pkg/apis/devices/v1alpha2/register.go](https://github.com/kubeedge/kubeedge/pull/4017/files#diff-2a5d882eaa81f850553891fa2e14085876d105e757d64322c0f491fd234a03b9).

Reply:
that means we need to manually write two files: `doc.go` and `types.go` if we want to develop a new CRD?
And does every `doc.go` need to contain the below comments? I find some CRD `doc.go` include only part of the below content.
```
// Package v1alpha2 contains API Schema definitions for the devices v1alpha2 API group
// +k8s:openapi-gen=true
// +k8s:deepcopy-gen=package,register
// +k8s:conversion-gen=kubeedge/cloud/pkg/apis/devices
// +k8s:defaulter-gen=TypeMeta
// +groupName=devices.kubeedge.io
```

Reply:
Yes, `doc.go` and `types.go` are all we need to write.
In my mind, not all of these are necessary, and the following two are needed.
1. `+k8s:deepcopy-gen` is used to generate zz_generated.deepcopy.go
2. `+groupName` indicates the group name of this CRD

Something like:  
1. `+k8s:openapi-gen` is used by `openapi-gen`.
2. `+k8s:conversion-gen` is used by `conversion-gen` to generate conversion funcs.

I've not found any usage here, and just respect the old code.

Reply:
ok, thanks for your explanation :)